resolver/dns/dns_resolver.go

```diff
@@ -24,10 +24,29 @@
 package dns
 
 import (
+	"time"
+
 	"google.golang.org/grpc/internal/resolver/dns"
 	"google.golang.org/grpc/resolver"
 )
 
+// SetResolvingTimeout sets the maximum duration for DNS resolution requests.
+//
+// This function updates the global `ResolvingTimeout` variable, which specifies
+// how long a DNS resolution attempt will wait before it is canceled.
+//
+// It is recommended to call this function at application startup, before any
+// gRPC calls are made. Modifying this value after initialization is not
+// thread-safe.
```

Comment:
Please replace "It is recommended to call this function..." with "This function must be called only...".